File: 2/main.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        dummy = ListNode(0)
        result = dummy
        Sum = 0
        
        while l1 or l2:
            if l1:
                Sum += l1.val
                l1 = l1.next
            if l2:
                Sum += l2.val
                l2 = l2.next
            result.next = ListNode(Sum % 10)
            result = result.next
            Sum //= 10
            
        if Sum > 0:
            result.next = ListNode(Sum);
            
        return dummy.next
        

File: 2/main.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode *dummy = new ListNode(0);
        ListNode *result = dummy;
        int sum = 0;
        
        while(l1 || l2) {
            if(l1) {
                sum += l1->val;
                l1 = l1->next;
            }
            if(l2) {
                sum += l2->val;
                l2 = l2->next;
            }
            result->next = new ListNode(sum % 10);
            result = result->next;
            sum /= 10;
        }
        
        if(sum > 0) {
            result->next = new ListNode(sum);
        }
        
        return dummy->next;
    }
};
